src/app/services/auth.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/compat/auth';
+import { BehaviorSubject, Observable, Subscriber } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  private currentUserSubject: BehaviorSubject<any>;
```

Comment:
You can define type and create BehaviorSubject here. Instead of creating on constructor function.
Like this
private currentUserSubject: BehaviorSubject<User> = new BehaviorSubject();

Or you can assign after the user information fetched.